classes/shop/Shop.php

```diff
@@ -464,13 +465,36 @@ public function getAddress()
     }
 
     /**
-     * Get shop theme name
-     *
-     * @return string
+     * Set shop theme details from Json data
      */
+    public function setTheme()
+    {
+        $dir = _PS_ALL_THEMES_DIR_.$this->theme_name;
+        $theme_data = Yaml::parse(file_get_contents(
```

Comment:
Also, I guess Yaml::parse is deprecated in Sf 2.8 (and removed in 3.0), you need to call Parser instance => http://symfony.com/doc/current/components/yaml/introduction.html#reading-yaml-files